persist/fs/commitlog/writer.go

```diff
@@ -90,7 +91,7 @@ type writer struct {
 	chunkReserveHeader []byte
 	buffer             *bufio.Writer
 	sizeBuffer         []byte
-	seen               bitSet
+	seen               *xsets.BitSet
```

Comment:
were you worried about the performance impact of having an interface here, or it just wasn't necessary?

Reply:
I was thinking that this might get moved to `m3x` at some point and if so to help avoid breaking changes when we add methods to things it might be worth keeping the pretty simple data structures that don't need to get mocked out as just structs.  

So this choice was mainly for reducing cross repo breaking changes.  

If it was a more complicated/logic based structured then we probably would've want to mock it.